webdictionary/src/main/java/com/example/webdictionary/NetworkConnection.java

```diff
@@ -0,0 +1,281 @@
+     */
+    public void addToNet(SMSPeer peer){
+        if(peer == null || !peer.isValid()) throw new IllegalArgumentException();
+        if(!subscribers.contains(peer)) subscribers.add(peer);
+    }
+
+    /**
+     * Adds a given array of valid peers to the network
+     * @param peers the SMSPeer array to add to the net
+     * @throws IllegalArgumentException If at least one peer is null or invalid,
+     * or if the array is null
+     */
+    public void addToNet(SMSPeer[] peers){
+        if(peers == null) throw new IllegalArgumentException();
+        for(SMSPeer peer: peers) addToNet(peer);
+    }
+    //endregion
+
+    //region removeFromNet
+    /**
+     * Removes a given String list of peers from the current network
+     * @param peersInNet a space separated string of valid SMSPeers
+     * @throws IllegalArgumentException if peersInNet is null or if at least
+     * one SMSPeer is null or invalid
+     */
+    public void removeFromNet(String peersInNet){
+        if(peersInNet == null) throw new IllegalArgumentException();
+        Log.d(LOG_KEY, "Removing these Peers: " + peersInNet);
+        String[] peers = peersInNet.split(" ");
+        for(String peer : peers){
+            removeFromNet(new SMSPeer(peer));
+        }
+    }
+
+    /**
+     * Removes a given SMSPeer from the current network
+     * @param peer The SMSPeer to remove from the net
+     * @throws IllegalArgumentException if peer is null or invalid
+     */
+    public void removeFromNet(SMSPeer peer){
+        if(peer == null || !peer.isValid()) throw new IllegalArgumentException();
+        Log.d(LOG_KEY, "Removing this Peer: " + peer);
+        subscribers.remove(peer);
+    }
+
+    /**
+     * Removes an array of valid SMSPeers from the current network
+     * @param peers The list of Peers to remove
+     * @throws IllegalArgumentException if peers is null, or if at least one SMSPeer
+     * is null or invalid
+     */
+    public void removeFromNet(SMSPeer[] peers){
+        if(peers == null) throw new IllegalArgumentException();
+        for (SMSPeer peer : peers) removeFromNet(peer);
+    }
+    //endregion
+
+    //region helperMethods
+    /**
+     * Removes every online peer from the net
+     * N.B. it doesn't notify anyone about the net, just clears it
+     */
+    public void clearNet(){
+        subscribers.clear();
+    }
+
+    /**
+     * Returns true if there is no online Peer in the net
+     */
+    public boolean isNetEmpty(){
+        return subscribers.isEmpty();
+    }
+
+    /**
+     * Returns how many SMSPeers are currently online
+     */
+    public int networkSize(){
+        return subscribers.size();
+    }
+
+    /**
+     * Returns an array of currently online SMSPeers
+     */
+    public SMSPeer[] getOnlinePeers(){
+        return subscribers.toArray(new SMSPeer[0]);
+    }
+
+    //endregion
+
+    /**
+     * Updates the current Network State given a peer in the network to update and it's resources
+     */
```

Comment:
This  is not very clear, i think you can rewrite it in a better way

Reply:
Yup, now that I read it I can see how it can get confusing, I'll quickly fix it!

Reply:
Thanks to your comment I was able to see that the function should not work this way, so some changes are needed on my part. For now you can see how the function specifics have changed